src/networking.c

```diff
@@ -209,6 +210,94 @@ cc_oci_tap_create(const gchar *const tap) {
 }
 
 /*!
+ * Set MTU for network interface.
+ *
+ * \param[in] ifname Interface name.
+ * \param mtu MTU to be set for the interface.
+ *
+ * \return \c true on success, else \c false.
+ */
+gboolean
+set_interface_mtu(char *ifname, unsigned int mtu)
+{
+	int socket_fd = -1;
+	gboolean ret = false;
+	struct ifreq ifr = { 0 };
+
+	if (! ifname) {
+		return false;
+	}
+
+	socket_fd = socket(AF_INET, SOCK_DGRAM, 0);
+	if (socket_fd < 0) {
+		g_critical("Error opening socket: %s", strerror(errno));
+		goto out;
+	}
+
+	ifr.ifr_addr.sa_family = AF_INET;
+	strncpy(ifr.ifr_name, ifname, IFNAMSIZ-1);
+	ifr.ifr_mtu = (int)mtu;
+
+	if (ioctl(socket_fd, SIOCSIFMTU, (caddr_t)&ifr) < 0) {
+		g_critical("ioctl SIOCSIFMTU (set mtu) error :%s", strerror(errno));
```

Comment:
I'd add `ifname` and `mtu` to this error message (alternatively, move the `g_debug()` call to the top of the function).